packages/ai-jsx/src/core/image-gen.tsx

```diff
@@ -77,14 +79,45 @@ export function ImageGenProvider<T extends ImageGenPropsWithChildren>(
  *    </ImageGen>
  * ```
  */
-export function ImageGen(
-  { children, ...props }: ImageGenPropsWithChildren & Record<string, unknown>,
-  { getContext }: RenderContext
-) {
+export function ImageGen({ children, ...props }: ImageGenPropsWithChildren, { getContext }: RenderContext) {
   const [ImageGenComponent, defaultProps] = getContext(imageGenContext);
   return (
     <ImageGenComponent {...defaultProps} {...props}>
       {children}
     </ImageGenComponent>
   );
 }
+
+/**
+ * A wrapper around {@link ImageGen} that renders the generated image(s) as as HTML "img" tag.
+ *
+ * @returns HTML "img" tag(s) with the generated image(s).
+ */
+export async function ImageGenHTML(
+  {
+    children,
+    width = undefined,
+    height = undefined,
+    alt = undefined,
+    ...props
+  }: ImageGenPropsWithChildren & {
+    /** Width for the HTML image tag. */ width?: string;
+    /** Height for the HTML image tag. */ height?: string;
+    /** Alternative text for the HTML image tag.  */ alt?: string;
+  },
+  { render }: RenderContext
+) {
+  const urls = await render(<ImageGen {...props}>{children}</ImageGen>);
+  return (
+    <AIR.React>
+      {urls.split('\n').map((url) =>
+        ReactModule.createElement('img', {
+          src: url,
+          alt: alt ?? 'Generated image',
+          width: width ?? (props.size ? props.size.split('x')[0] : undefined),
+          height: height ?? (props.size ? props.size.split('x')[1] : undefined),
+        })
+      )}
+    </AIR.React>
```

Comment:
Any thoughts on whether this use is Kosher @NickHeiner?

Reply:
I might call it `AIReact` but this is internal code so nbd.